docs/walkthroughs/07-enabling-collaborative-editing.md

```diff
@@ -0,0 +1,397 @@
+    setSharedType(sharedDoc)
+    setProvider(yProvider)
+
+    return () => {
+      yDoc?.destroy()
+      yProvider?.off('sync', setConnected)
+      yProvider?.destroy()
+    }
+  }, [])
+
+  if (!connected || !sharedType || !provider) {
+    return <div>Loading…</div>
+  }
+
+  return <SlateEditor />
+}
+
+const SlateEditor = () => {
+  const [editor] = useState(() => withReact(createEditor()))
+
+  return (
+    <Slate editor={editor} initialValue={initialValue}>
+      <Editable />
+    </Slate>
+  )
+}
+```
+
+After setting up your Yjs document like this, you can then link it your editor by passing down `sharedType`, which
+contains the multiplayer text, and by using functions from `slate-yjs`. We're also passing down `provider` which will be
+helpful later.
+
+```jsx
+import { useEffect, useMemo, useState } from 'react'
+import { createEditor, Editor, Transforms } from 'slate'
+import { Editable, Slate, withReact } from 'slate-react'
+import { withYjs, YjsEditor } from '@slate-yjs/core'
+import * as Y from 'yjs'
+
+const initialValue = {
+  children: [{ text: '' }],
+}
+
+export const CollaborativeEditor = () => {
+  const [connected, setConnected] = useState(false)
+  const [sharedType, setSharedType] = useState()
+  const [provider, setProvider] = useState()
+
+  // Connect to your Yjs provider and document
+  useEffect(() => {
+    const yDoc = new Y.Doc()
+    const sharedDoc = yDoc.get('slate', Y.XmlText)
+
+    // Set up your Yjs provider. This line of code is different for each provider.
+    const yProvider = new YjsProvider(/* ... */)
+
+    yProvider.on('sync', setConnected)
+    setSharedType(sharedDoc)
+    setProvider(yProvider)
+
+    return () => {
+      yDoc?.destroy()
+      yProvider?.off('sync', setConnected)
+      yProvider?.destroy()
+    }
+  }, [])
+
+  if (!connected || !sharedType || !provider) {
+    return <div>Loading…</div>
+  }
+
+  return <SlateEditor sharedType={sharedType} provider={provider} />
+}
+
+const SlateEditor = ({ sharedType, provider }) => {
+  const editor = useMemo(() => {
+    const e = withReact(withYjs(createEditor(), sharedType))
+
+    // Ensure editor always has at least 1 valid child
+    const { normalizeNode } = e
+    e.normalizeNode = entry => {
+      const [node] = entry
+
+      if (!Editor.isEditor(node) || node.children.length > 0) {
+        return normalizeNode(entry)
+      }
+
+      Transforms.insertNodes(editor, initialValue, { at: [0] })
+    }
+
+    return e
+  }, [])
+
+  useEffect(() => {
+    YjsEditor.connect(editor)
+    return () => YjsEditor.disconnect(editor)
+  }, [editor])
+
+  return (
+    <Slate editor={editor} initialValue={initialValue}>
+      <Editable />
+    </Slate>
+  )
+}
+```
+
+That's all you need to attach Yjs to Slate!
+
+Let's look at a real-world example of setting up Yjs—here's a code snippet for setting up
+a [Liveblocks provider](https://liveblocks.io/docs/get-started/yjs-slate-react). Liveblocks uses the concept of rooms,
```

Comment:
I would like to see a short disclaimer along the lines of (but maybe something better than my draft):

"Liveblocks is a commercial service independent of the Slate project. It is not required for enabling collaboration with Slate or Yjs, but may be a convenient option depending on your needs"

Reply:
Good idea! I'll link to other providers too.